extension/intents/forms/forms.js

```diff
@@ -72,3 +85,25 @@ intentRunner.registerIntent({
     await browser.tabs.sendMessage(activeTab.id, { type: "focusPrevious" });
   },
 });
+
+intentRunner.registerIntent({
+  name: "forms.turnSelectionIntoLink",
+  async run(context) {
+    const activeTab = await context.activeTab();
+    const selection = await pageMetadata.getSelection(activeTab.id);
+    if (!selection || !selection.text) {
+      const e = new Error("No text selected");
+      e.displayMessage = "No text selected";
+      throw e;
+    }
+    const url = createTabGoogleLucky(selection.text);
+    await content.lazyInject(activeTab.id, [
+      "/js/vendor/fuse.js",
+      "/intents/forms/formsContentScript.js",
+    ]);
+    await browser.tabs.sendMessage(activeTab.id, {
+      type: "turnSelectionIntoLink",
+      url,
+    });
```

Comment:
Once everything is done you can close the tab you opened above.

Reply:
@ianb  sorry as i have misunderstood the things as horribly wrong. After updating the changes suggested by you I am still getting some problems. Like here the `newTab.title` is not giving an appropriate tiltle and seems to give something like url

Reply:
This should still close. I.e., call `browser.tabs.remove(newTab.id)`